src/org/jetbrains/plugins/scala/caches/HydraArtifactsCache.scala

```diff
@@ -1,23 +1,28 @@
 package org.jetbrains.plugins.scala.caches
 
 import java.io.File
-import java.util.concurrent.ConcurrentHashMap
+
 import org.jetbrains.plugins.scala.project.template.FileExt
 import com.intellij.openapi.diagnostic.Logger
 import org.jetbrains.plugins.scala.project.template.Downloader
+import org.jetbrains.plugins.scala.settings.HydraApplicationSettings
 
 /**
   * @author Maris Alexandru
   */
 object HydraArtifactsCache {
   private val Log = Logger.getInstance(this.getClass)
-  private val cache = new ConcurrentHashMap[(String, String), Seq[String]]()
+  private val hydraGlobalSettings = HydraApplicationSettings.getInstance()
   private val SplitRegex = "\\* Attributed\\(|\\)".r
   private val ArtifactsRegex = "\\* Attributed\\(.*\\)".r
   private val GroupId = "com.triplequote"
 
-  def getOrDownload(scalaVersion: String, hydraVersion: String, listener: (String) => Unit): Seq[String] = {
-    val artifacts = cache.getOrDefault((scalaVersion, hydraVersion), Seq.empty)
+  def checkIfArtifactsExist(artifacts: List[String]): Boolean = {
+    artifacts.map(new File(_)).forall(_.exists())
```

Comment:
Small point, but I'd avoid the extra iteration, i.e., `artifacts.forall(new File(_).exists())`